shelley-ma/formal-spec/value.tex

```diff
@@ -28,46 +26,43 @@ \section{Coin and Multi-Asset Tokens}
       \var{quan} & \Quantity & \Z \\
       %\text{quantity of a token}\\
       \var{v}, \var{w} & \Value
-      & \PolicyID \to_0 ( \AssetID \to_0 \Quantity )
+      & \PolicyID \to_0 ( \AssetName \to_0 \Quantity )
 %      & \text{a collection of tokens}
     \end{array}
   \end{equation*}
   %
   \emph{Abstract Functions and Values}
   %
   \begin{align*}
-    \mathsf{adaID} \in& ~\PolicyID & \text{Ada asset ID} \\
-    \mathsf{adaToken} \in& ~\AssetID & \text{Ada Token} \\
-    \fun{valueSize} \in& ~\Value \to \N & \text{Memory estimate for}~\Value
+    \mathsf{adaID} \in& ~\PolicyID & \text{Ada Policy ID} \\
+    \mathsf{adaName} \in& ~\AssetName & \text{Ada Asset Name} \\
   \end{align*}
   \emph{Helper Functions}
   %
   \begin{align*}
     & \fun{coinToValue} \in \Coin\to \Value \\
-    & \fun{coinToValue}~c = \lambda~\var{pid}~\var{aid}.
+    & \fun{coinToValue}~c = \lambda~\var{pid}~\var{aname}.
       \begin{cases}
-        c & \var{pid} = \mathsf{adaID}~\text{and}~\var{aid} = \mathsf{adaToken} \\
+        c & \var{pid} = \mathsf{adaID}~\text{and}~\var{aname} = \mathsf{adaName} \\
         0 & \text{otherwise}
       \end{cases}
     \nextdef
     & \fun{valueToCoin} \in \Value \to \Coin \\
-    & \fun{valueToCoin}~v = v~\mathsf{adaID}~\mathsf{adaToken}
+    & \fun{valueToCoin}~v = v~\mathsf{adaID}~\mathsf{adaName}
   \end{align*}
   \caption{Type Definitions and auxiliary functions for Value}
   \label{fig:defs:value}
 \end{figure*}
 
 \subsection*{Representing Multi-asset Types and Values}
 An \emph{Asset} comprises a set of different \emph{Asset Classes}, each of which has
-a unique identifier, $\AssetID$. We will informally refer to a pair $(\var{pid}, \var{aid})$ of a Policy ID and an Asset ID as a ``token''. The token for Ada is $(\mathsf{adaID}, \mathsf{adaToken})$.
+a unique identifier, $\AssetName$. We will refer to a pair of a $\PolicyID$ and an
+$\AssetName$ as an $\AssetID$. The $\AssetID$ for Ada is $(\mathsf{adaID}, \mathsf{adaName})$.
```

Comment:
should \mathsf{adaID} be \mathsf{adaPolicy}, then?

Reply:
Makes sense!